implementations/rust/ockam/ockam_vault_aws/src/vault/aws_kms_client.rs

```diff
@@ -269,30 +271,30 @@ fn digest(data: &[u8]) -> Blob {
 #[derive(Error, Debug)]
 pub(crate) enum Error {
     #[error("aws sdk error creating new key")]
-    Create(#[from] SdkError<CreateKeyError>),
+    Create(#[from] SdkError<CreateKeyError, Response<SdkBody>>),
```

Comment:
That would mean converting the whole `SdkError<CreateKeyError, Response<SdkBody>>` to a `String` because the second type parameter in `SdkError` has to be a `Response`. I'd rather keep the whole type then.

Reply:
As far as I understand: this enum is crate-only. Therefore, we only use it from outside by casting it to `ockam_core::Error`. After that cast, it's impossible to recover any of these structures, you can only print the error. I also haven't noticed these error details being used from inside the crate (I may be wrong). Therefore, we could safely cast it to the string in the very beginning. That should allow us to remove `aws-smithy-http` and `http` from our `Cargo.toml`, which should be desirable for us (having a smaller number of abstractions, versions, etc. to manage). They would still be transitive dependencies though.
That all being said, I'm fine with merging it as it is, until we find time to rethink our Error approach overall.

Reply:
Thanks!